create_venv.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+The script can be used to setup a virtual environment for running Firefox UI Tests.
+It will automatically install the firefox ui test package, all its dependencies, and
+auxiliary packages like mozdownload.
+"""
+
+import optparse
+import os
+import shutil
+import subprocess
+import sys
+import urllib2
+import zipfile
+
+
+# Link to the folder, which contains the zip archives of virtualenv
+VIRTUALENV_URL = 'https://github.com/pypa/virtualenv/archive/%(VERSION)s.zip'
+VIRTUALENV_VERSION = '12.1.1'
+
+here = os.path.dirname(os.path.abspath(__file__))
+
+venv_script_path = 'Scripts' if sys.platform == 'win32' else 'bin'
+venv_activate = os.path.join(venv_script_path, 'activate')
+venv_activate_this = os.path.join(venv_script_path, 'activate_this.py')
+venv_python_bin = os.path.join(venv_script_path, 'python')
+
+usage_message = """
+***********************************************************************
+To run the Firefox UI Tests, activate the virtual environment:
+    {}{}
+
+See firefox-ui-tests --help for all options
+
+***********************************************************************
+"""
+
+
+def download(url, target):
+    """Downloads the specified url to the given target."""
+    response = urllib2.urlopen(url)
+    with open(target, 'wb') as f:
+        f.write(response.read())
+
+    return target
+
+
+def create_virtualenv(target, python_bin=None):
+    script_path = os.path.join(here, 'virtualenv-%s' % VIRTUALENV_VERSION,
+                               'virtualenv.py')
+
+    print 'Downloading virtualenv %s' % VIRTUALENV_VERSION
+    zip_path = download(VIRTUALENV_URL % {'VERSION': VIRTUALENV_VERSION},
+                        os.path.join(here, 'virtualenv.zip'))
+
+    try:
+        with zipfile.ZipFile(zip_path, 'r') as f:
+            f.extractall(here)
+
+        print 'Creating new virtual environment'
+        cmd_args = [sys.executable, script_path, target]
+
+        if python_bin:
+            cmd_args.extend(['-p', python_bin])
+
+        subprocess.check_call(cmd_args)
+    finally:
+        try:
+            os.remove(zip_path)
+        except OSError:
+            pass
+
+        shutil.rmtree(os.path.dirname(script_path), ignore_errors=True)
+
+
+def main():
+    parser = optparse.OptionParser('Usage: %prog [options] path_to_venv')
+    parser.add_option('-p', '--python',
+                      type='string',
+                      dest='python',
+                      metavar='BINARY',
+                      help='The Python interpreter to use.')
+    parser.add_option('--with-requirements',
+                      dest='with_requirements',
+                      action='store_true',
+                      help='Install optional packages as listed in requirements.txt')
```

Comment:
It's funny to refer to these as "optional requirements". I don't have a problem making pep8 and mozdownload requirements for the firefox-ui-tests, but if not maybe find a way to refer to them that indicates they're for setting up the environment rather than running the tests themselves.